custom/icds_reports/tasks.py

```diff
@@ -1199,6 +1199,46 @@ def build_incentive_report(agg_date=None):
         agg_date = current_month - relativedelta(months=1)
     for state in state_ids:
         AWWIncentiveReport.aggregate(state, agg_date)
+
+    awc_performance_rows_list = list(AWWIncentiveReport.objects.values_list('awc_id', 'is_launched',
+                                                                            'valid_visits', 'visit_denominator',
+                                                                            'wer_weighed', 'wer_eligible',
+                                                                            'incentive_eligible')
+                                     .order_by('awc_id'))
+    awc_aggregate_rows_list = list(AggAwc.objects.values_list('awc_id', 'is_launched',
+                                                              'valid_visits', 'expected_visits')
+                                   .order_by('awc_id'))
+    is_launched_check_bad_data = []
+    home_conduct_check_bad_data = []
+    eligibility_check_bad_data = []
+    for index, awc_from_performance in enumerate(awc_performance_rows_list):
+        awc_from_aggregate = awc_aggregate_rows_list[index]
+
+        # check for launched AWCs
+        row_data = get_awc_is_launched_mismatch_row(awc_from_performance, awc_from_aggregate)
+        if row_data is not None:
+            is_launched_check_bad_data.append(row_data)
+
+        # check for home conduct percentage
+        row_data = get_awc_home_conduct_mismatch_row(awc_from_performance, awc_from_aggregate)
+        if row_data is not None:
+            home_conduct_check_bad_data.append(row_data)
+
+        # check for eligibility
+        row_data = get_awc_eligibility_mismatch_row(awc_from_performance)
+        if row_data is not None:
+            eligibility_check_bad_data.append(row_data)
+
+    # sending email with is_launched mismatch csv
+    if len(is_launched_check_bad_data) > 0:
+        awc_is_launched_mismatch_mail_helper(is_launched_check_bad_data)
+    # sending email with home conduct % mismatch csv
+    if len(home_conduct_check_bad_data) > 0:
+        awc_home_conduct_mismatch_mail_helper(home_conduct_check_bad_data)
+    # sending email with eligibility mismatch csv
+    if len(eligibility_check_bad_data) > 0:
+        awc_eligibility_mismatch_mail_helper(eligibility_check_bad_data)
```

Comment:
I think I might be missing something in the comment. This is being called async in a separate celery task so that it wont block the generation. At this point, that is to prevent a false positive from breaking the report in addition to the point you made about not wanting things like email failure to be blocking.

Reply:
hey, I made this comment on the initial stage of the PR when all the checking was happening with sequential manner in same function. Now that after its made async, so, all good now for me.